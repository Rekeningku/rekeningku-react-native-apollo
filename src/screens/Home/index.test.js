import React from 'react'
import sinon from 'sinon'
import { shallow } from 'enzyme'
import { HomeScreen } from './index'


describe('Home Component', () => {
  it('renders with all props', () => {
    const wrapper = shallow(<HomeScreen />)
    expect(wrapper).toMatchSnapshot()
  })

  it('calls open screen handler', () => {
    const handleOpenScreenA = sinon.fake()
    const wrapper = shallow(
      <HomeScreen handleOpenScreenA={handleOpenScreenA} />,
    )
    wrapper.find('ButtonComponent').simulate('press')
    expect(handleOpenScreenA.callCount).toBe(1)
  })
})
