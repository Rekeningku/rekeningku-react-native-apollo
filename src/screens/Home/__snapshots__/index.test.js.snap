// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home Component renders with all props 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <HomeScreen />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <Styled(Component)>
          Home Screen
        </Styled(Component)>,
        <Styled(Component)>
          This is an example screen that is not connected to redux-store.
        </Styled(Component)>,
        <Styled(Component)>
          With Wix React Native Navigation it is easy to make native transitions between screens.
        </Styled(Component)>,
        <ButtonComponent>
          Open Nested Screen A
        </ButtonComponent>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": "Home Screen",
        },
        "ref": null,
        "rendered": "Home Screen",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": "This is an example screen that is not connected to redux-store.",
        },
        "ref": null,
        "rendered": "This is an example screen that is not connected to redux-store.",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": "With Wix React Native Navigation it is easy to make native transitions between screens.",
        },
        "ref": null,
        "rendered": "With Wix React Native Navigation it is easy to make native transitions between screens.",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": "Open Nested Screen A",
          "onPress": undefined,
        },
        "ref": null,
        "rendered": "Open Nested Screen A",
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <Styled(Component)>
            Home Screen
          </Styled(Component)>,
          <Styled(Component)>
            This is an example screen that is not connected to redux-store.
          </Styled(Component)>,
          <Styled(Component)>
            With Wix React Native Navigation it is easy to make native transitions between screens.
          </Styled(Component)>,
          <ButtonComponent>
            Open Nested Screen A
          </ButtonComponent>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "Home Screen",
          },
          "ref": null,
          "rendered": "Home Screen",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "This is an example screen that is not connected to redux-store.",
          },
          "ref": null,
          "rendered": "This is an example screen that is not connected to redux-store.",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "With Wix React Native Navigation it is easy to make native transitions between screens.",
          },
          "ref": null,
          "rendered": "With Wix React Native Navigation it is easy to make native transitions between screens.",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": "Open Nested Screen A",
            "onPress": undefined,
          },
          "ref": null,
          "rendered": "Open Nested Screen A",
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
